vnext/local-cli/config/dependencyConfig.js

```diff
@@ -1,102 +1,117 @@
-const fs = require('fs');
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * @format
+ */
+// @ts-check
+
 const path = require('path');
-const glob = require('glob');
-const xmldoc = require('xmldoc');
 
-function dependencyConfigWindows(folder, userConfig = {}) {
-  const sourceDir = userConfig.sourceDir || findWindowsAppFolder(folder);
+const configUtils = require('./configUtils.js');
+
+/*
+
+Schema for dependencies:
+
+{
+  folder: '', // absolute path to the module root, ex: c:\path\to\node_modules\module-name
+  sourceDir: '', // relative path to the windows implementation folder, ex: windows
+  solutionFile: '', // relative path to the solution, ex: ModuleName.sln
+  projects: [ // array of projects to be built that include modules
+    {
+      projectFile: '', // relative path to the project, ex: ProjectName\ProjectName.vcxproj
+      projectName: '', // name of the project
+      projectLang: '', // language of the project, cpp or cs
+      projectGuid: '', // project identifier
+      cppHeaders: [], // array of cpp header include lines, ie: 'winrt/ModuleName.h', to be transformed into '#include <winrt/ModuleName.h>'
+      cppPackageProviders: [], // array of fully qualified cpp IReactPackageProviders, ie: 'ModuleName::ReactPackageProvider'
+      csNamespaces: [], // array of cs namespaces, ie: 'ModuleName', to be transformed into 'using ModuleName;'
+      csPackageProviders: [], // array of fully qualified cs IReactPackageProviders, ie: 'ModuleName.ReactPackageProvider'
+    },
+  ],
+  additionalProjects: [ // array of (dependency) projects that must be built, but do not contain modules
+    {
+      projectFile: '', // relative path to the project, ex: ProjectName\ProjectName.vcxproj
+      projectName: '', // name of the project
+      projectLang: '', // language of the project, cpp or cs
+      projectGuid: '', // project identifier
+    },
+  ],
+  packages: [ // array of nuget packages that include modules
```

Comment:
This should be called nugetPackages or something.  RN developers are going to think NPM packages when they just see packages.

And this whole thing is just needed because VS C++ projects can't include nuget references directly right?  I'm worried that we are kinda trying to implement a VS feature here..  Shouldn't we just get VS to fix C++ nuget support?  And if they dont.. then I'm not sure we should take that on.   Also why does this section need another set of cppPackageProviders etc?  Can't they just be specified by one of the projects in the first list?  (projects.cppPackageProviders)

Reply:
No, this is to support the scenario where a native module publishes to npm and chooses not to publish windows source in their npm package, but instead publishes a nuget binary. The react-native config can't possibly figure that out on it's own, so they'll have to create a react-native.config file that specifies the nuget package name and version string that matches npm package.

Autolinking will then read this info and use it to either add packagereferences in the generated targets file (for C#) or modify the app's packages.config (for C++).

I can change the name to nuget package to make it more clear.

Reply:
It at some point VS does support packagereferences for C++, then we just update auto-linking to add packagereferences for both C# and C++ and stop touching packages.config.

Reply:
Fixed.